src/LaravelScoutExtendedServiceProvider.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of Laravel Scout Extended.
+ *
+ * (c) Algolia Team <contact@algolia.com>
+ *
+ *  For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ */
+
+namespace Algolia\LaravelScoutExtended;
+
+use Laravel\Scout\Builder;
+use Laravel\Scout\EngineManager;
+use Algolia\AlgoliaSearch\Client;
+use Algolia\AlgoliaSearch\Places;
+use Algolia\AlgoliaSearch\Analytics;
+use Illuminate\Support\ServiceProvider;
+use Laravel\Scout\Engines\AlgoliaEngine;
+use Algolia\AlgoliaSearch\Interfaces\ClientInterface;
+
+final class LaravelScoutExtendedServiceProvider extends ServiceProvider
```

Comment:
We should add a UserAgent segment to pass the package name and version. I would typically do it in the register method here. What do you think?

Reply:
Humm. Can you give me a practical example that explains the advantages for the end-developer?

Reply:
The UserAgent is used to pass the version of the API Clients and integration. It helps us understand what framework versions, PHP versions are used by our users. This is typically how we decided to maintain PHP 5.3 for the new PHP Clients.
The Algolia display this information to the user and it will also alert you if you're using an API client that is too old or have known issues.

Reply:
@julienbourdeau Understood! Thanks for the help. 🤗